zerver/actions/message_edit.py

```diff
@@ -1003,6 +1031,13 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
                     visibility_policy=new_visibility_policy,
                 )
 
+    if topic_name is not None and not target_topic_has_messages:
```

Comment:
This doesn't work if we are changing the stream only & topic name is kept the same.